frontend/src/pages/AccountSend.vue

```diff
@@ -398,6 +409,21 @@ function useSendForm() {
     sendFormRef.value?.resetValidation();
   }
 
+  async function setHumanAmountMax() {
+    if (!token.value?.address) throw new Error(vm.$i18n.tc('Send.select-a-token'));
+    if (NATIVE_TOKEN.value?.address === token.value?.address) {
+      const address = userAddress.value;
+      if (!address || !provider.value) throw new Error(vm.$i18n.tc('Send.wallet-not-connected'));
+      if (!recipientId.value) throw new Error(vm.$i18n.tc('Send.enter-a-recipient'));
+      const { ethToSend } = await umbraUtils.getEthSweepGasInfo(address || '', recipientId.value, provider.value);
```

Comment:
I swear I left a comment about this line but apparently not, sorry! 😅 Two things:
1. We can change `address || ''` to just `address`, since we throw an error a few lines up if it's undefined
2. Using an ENS name as the recipient like `msolomon.eth` results in an error, because `getEthSweepGasInfo` wants an address but we only give it a string. `address.ts` has a `toAddress` method you can use so this should be an easy fix with just `await toAddress(recipientId.value)`

Reply:
No worries, updated

Reply:
The current approach isn't exactly correct, sorry if I was unclear! Right now we have the below code, which I annotated with comments:

```typescript
// This line is basically a no-op, it inputs the user's address and returns it
const address = await toAddress(userAddress.value, provider.value);

// This line is "correct", but reverts because `recipientId.value` might be an ENS name.
// On mainnet, ethersjs will handle that automatically, but on L2s it won't.
const { ethToSend } = await umbraUtils.getEthSweepGasInfo(address, recipientId.value, provider.value);
```

What we want is something like this:

```typescript
// This is basically what you had before, which is correct, and I just renamed
// the variable name to make it more clear. I also added the `!` which tells
// Typescript that we are certain the address will be defined (since you have
// to connect a wallet to access this page, but TS doesn't know that)
const fromAddress = userAddress.value!;

// Now we take the recipient ID, i.e. who you are sending to, and make sure
// it's an address. This can be a string, like an ENS name, so we must resolve
// it to a standard address.
const toAddress = await toAddress(recipientId.value, provider.value);

// Now we calculate the sweep info, and the first two arguments are now
// standard addresses.
const { ethToSend } = await umbraUtils.getEthSweepGasInfo(fromAddress, toAddress, provider.value);
```

I'd recommend testing the functionality on a few networks, like mainnet and arbitrum, with both regular addresses and ENS names, to make sure it works as expected. An ENS name like `vitalik.eth` or `msolomon.eth` should work for all of them (except rinkeby—on live networks, we also resolve ENS names against mainnet, but on rinkeby (since it's a testnet) we resolve directly on rinkeby).

If you haven't already, you can also use your own ENS name to go through the full cycle of sending the max to yourself, then withdrawing to yourself, to both test the send process but also to have a better understanding of the full flow 🙂 (ideally with rinkeby to avoid losing real funds if the PR doesn't work as intended)

Reply:
Oh sorry, I misunderstood. That makes sense, thank you for the thorough explanation. I will resolve this

Reply:
Updated, and tested sending to your ENS on rinkeby

Reply:
Hmm, I did get your rinkeby ETH, but I'm not sure how it worked since we're now not using `recipientId.value` at all—it seems the recipient address is now set to the sender's address and not whatever `recipientId.value` resolves to?

https://github.com/alexkeating/umbra-protocol/blob/1bd08cd3f4ca09efe260cfb6613e22fd342d344a/frontend/src/pages/AccountSend.vue#L416-L417

Reply:
Yes, that is weird. I added replaced the offending line with recipientId.value

Reply:
awesome, thanks! pushed one commit to refactor the checks logic a bit to make it a little cleaner, but looks great now, thakns!